### PR TITLE
Add hypercharge mode for bounties

### PR DESCRIPTION
**Pull request by zoek1**
##### Description

Admin flag on bounties where someone can set a bool "hypercharge mode" and save
When a bounty is in hypercharge mode (ie. it is not done/cancelled), it will be

- [X] tweeted 2x/day automatically
- [X] gitcoinbot will write a pinned post about it
image

- [X] there will be secret/daily offers for the bounty
image

- [X] The bounty will be featured in the daily email.
image

- [X] Notify users who have worked with that org before
image

- [X] Auto feature on hack explorer/normal prize explorer
image

##### Refers/Fixes
https://github.com/gitcoinco/web/issues/7135

### CODE REVIEW

File: app/dashboard/models.py [hunk 1]
```diff
@@ -1299,6 +1305,43 @@ def total_reserved_length_label(self):
...
+                'title': title,
+                'description': truncatechars(instance.issue_description_text, 500),
+                'url': instance.get_absolute_url(),
+                'ask': '#announce'
+        }
+        activity = Activity.objects.create(profile=profile, activity_type='hypercharge_bounty',
+                                           metadata=metadata, bounty=instance)
+
+        PinnedPost.objects.filter(what='everywhere').delete()
+        # activity = Activity.objects.get(bounty=instance)
```
Comment (danlipert) [inline app/dashboard/models.py:1333]: leftover comment

File: app/marketing/management/commands/hypercharge_mode.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+from datetime import datetime
+
+from django.core.management.base import BaseCommand
+from django.template.defaultfilters import floatformat
+from django.utils import timezone
+
+from dashboard.models import Activity, Earning, Profile, Bounty, Interest, BountyEvent
+from dashboard.notifications import maybe_market_to_twitter
+from economy.utils import convert_token_to_usdt
+from grants.models import *
```
Comment (danlipert) [inline app/marketing/management/commands/hypercharge_mode.py:33]: Just import what you need here

File: app/retail/templates/emails/bounty_hypercharged.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+<p>Work on "{{ bounty.title }}" and receive {{ bounty.value_true|floatformat }} {{ bounty.token_name }}</p>
+
+<p>{{ bounty.issue_description_text|truncatechars:500 }}</p>
+
+<div style="margin-bottom: 1em; margin-top: 1em;">
+  <a class="button" href="{{ bounty.absolute_url }}">{% trans "View on Gitcoin.co" %}</a>
+</div>
+
+<p>
+{% trans "You are receiving this email because your email address is on the notification list for this funded issue." %}  <a href="{% url 'email_settings' subscriber.priv %}">{% trans "Manage Email Settings" %}</a>.
```
Comment (danlipert) [inline app/retail/templates/emails/bounty_hypercharged.html:35]: Is this language regarding the notification list what we actually want to use? @PixelantDesign @frankchen07

Comment (owocki) [reply app/retail/templates/emails/bounty_hypercharged.html:35]: i can sub in the language we want to use

**Comment by owocki** [comment]
For the daily offer, can it pleast just tease the amount and title of the bounty in the desc? No need to pull the full unformatted markdown in there.

File: app/dashboard/models.py [hunk 4]
```diff
@@ -1299,6 +1305,43 @@ def total_reserved_length_label(self):
...
+
+        title = f'Work on "{instance.title}" and receive {floatformat(instance.value_true)} {instance.token_name}'
+
+        # Feature bounty on hackathon/prize explorer
+        instance.is_featured = True
+        instance.featuring_date = timezone.now()
+        instance.hyper_next_publication = timezone.now()
+
+        # Publish and pin on townsaquare
+        profile = Profile.objects.filter(handle='owocki').first()
```
Comment (owocki) [inline app/dashboard/models.py:1322]: ill probably wanna try/except this. not everyone has an  `owocki` on their local

**Comment by owocki** [comment]
looks really good; keen to get this live.  @zoek1 can u turn around @danlipert code review

**Comment by zoek1** [comment]
For sure @owocki, give me a moment

File: app/dashboard/models.py [hunk 5]
```diff
@@ -1318,22 +1318,22 @@ def post_save_bounty(sender, instance, created, **kwargs):
...
-        metadata = {
-                'title': title,
-                'description': truncatechars(instance.issue_description_text, 500),
-                'url': instance.get_absolute_url(),
-                'ask': '#announce'
-        }
-        activity = Activity.objects.create(profile=profile, activity_type='hypercharge_bounty',
-                                           metadata=metadata, bounty=instance)
+        # Publish and pin on townsquare
+        profile = Profile.objects.filter(handle='connoroday').first()
```
Comment (danlipert) [inline app/dashboard/models.py:1322]: Can we do a followup PR that allows this to be set in the DB OR via the app environment settings?

**Comment by octavioamu** [comment]
Can we just put the title instead of the content? seems we are sending all the template and sort of confusing + better to get people clicking and getting into the bounty.

File: app/dashboard/models.py (hunk 5 above)
Comment (zoek1) [reply app/dashboard/models.py]: i moved this to settings as env variable

**Comment by zoek1** [comment]
The email now display only the title :+1:  @octavioamu

**Pull request merged**